src/uu/expr/src/syntax_tree.rs

```diff
@@ -55,10 +56,10 @@ impl BinOp {
 }
 
 impl RelationOp {
-    fn eval(&self, a: &AstNode, b: &AstNode) -> ExprResult<String> {
+    fn eval(&self, a: &AstNode, b: &AstNode) -> ExprResult<NumOrStr> {
         let a = a.eval()?;
         let b = b.eval()?;
-        let b = if let (Ok(a), Ok(b)) = (a.parse::<BigInt>(), b.parse::<BigInt>()) {
+        let b = if let (NumOrStr::Num(a), NumOrStr::Num(b)) = (&a, &b) {
```

Comment:
This does not quite mean the same thing as before, because being a number is different from being a number or being a string representing a number.